src/rest_framework_dso/serializer_helpers.py

```diff
@@ -23,3 +25,16 @@ def __repr__(self):
     def __bool__(self):
         # Generators are always true, calls to "if data: ..." should not fail.
         return True
+
+
+def peek_iterable(generator):
```

Comment:
I'm a big fan of `more-itertools`. It has functions for looking ahead and back:

https://more-itertools.readthedocs.io/en/stable/api.html?highlight=peek#lookahead-and-lookback

Might want to consider using that.

Reply:
This is such small case, that I avoided bringing more-itertools in. Otherwise, yes, it would have been nice.